lms/services/organization.py

```diff
@@ -238,6 +239,40 @@ def update_organization(  # noqa: PLR0913
 
         return organization
 
+    def generate_usage_report(
+        self, organization_id: int, tag: str, since: datetime, until: datetime
+    ):
+        """Generate and store an usage report for one organization."""
+        organization = self.get_by_id(organization_id)
+        assert organization
+        report_key = OrganizationUsageReport.generate_key(
+            organization, tag, since, until
+        )
+
+        if (
+            report := self._db_session.query(OrganizationUsageReport)
+            .filter_by(key=report_key)
+            .one_or_none()
+        ):
+            LOG.debug("Report already exists, skipping generation. %s", report_key)
+            return report
+
+        report = OrganizationUsageReport(
+            organization=organization,
+            key=report_key,
+            since=since,
+            until=since,
```

Comment:
Is this intended, or is it a typo?

```suggestion
            until=until,
```

Reply:
Yes!, good catch. Added the missing test assertion for this.